.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,31 @@
+name: Node CI
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: npm install, build, and test
+        run: |
+          npm install
+          npm run build --if-present
+          npm test
+        env:
+          CI: true
```

Comment:
Is the matrix testing required for the eslint plugin? We already have the `pr-e2e.yml` action that runs tests on PRs, and then we have `deploy-next.yml` that runs the tests on pushes.

If this is only supposed to run on the eslint plugin it would need to use the [`--workspace` flag](https://docs.npmjs.com/cli/v8/using-npm/workspaces?v=true) and [filter via path](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore). 

As is, both actions will run tests for all packages that have a `test` npm script. Turbo's remote caching is used to make sure tests only run on the packages that were changed, but we haven't set that up yet since afaik it's a paid service.

Reply:
We could also rename the eslint plugin's `test` npm script so it doesn't get executed by turbo's `test` pipeline.